src/storage/admin/AdminTaskManager.cpp

```diff
@@ -116,7 +223,12 @@ void AdminTaskManager::schedule() {
           "job {}, genSubTask failed, err={}",
           task->getJobId(),
           apache::thrift::util::enumNameSafe(nebula::error(errOrSubTasks)));
+      saveTaskStatus(task->getJobId(),
+                     task->getTaskId(),
+                     task->cmdType(),
+                     nebula::meta::cpp2::JobStatus::FAILED);
       task->finish(nebula::error(errOrSubTasks));
+      removeTaskStatus(task->getJobId(), task->getTaskId());
```

Comment:
A question, whether job status is read in phase task->finish? if not, can we just delete this state? don't need to report the newly status via saveTaskStatus.

Reply:
> A question, whether job status is read in phase task->finish? if not, can we just delete this state? don't need to report the newly status via saveTaskStatus.

If we don't report the status,it will always be RUNNING in metad.